### PR TITLE
Removes hardcoded rollup genesis

### PR DESCRIPTION
**Pull request by otherview**
Requirement for https://github.com/obscuronet/obscuro-playground/pull/67

The first node on the network now creates the genesis based on the last block he read.
Following nodes will expect to find a genesis rollup on the blocks.

### CODE REVIEW

**Review by joeldudley — APPROVED** [review]
LGTM

File: go/obscuronode/enclave/state.go [hunk 1]
```diff
@@ -110,16 +110,28 @@ func updateState(b *types.Block, s Storage, blockResolver BlockResolver) *blockS
...
-			block:          b,
-			head:           &GenesisRollup,
-			state:          emptyState(),
-			foundNewRollup: true,
+	genesisRollup := s.FetchRollupGenesis()
+	if len(rollups) == 1 {
+		rol := rollups[0]
+
+		// the incoming block might hold the genesis rollup
+		// todo change this to an hardcoded hash on testnet/mainnet
```
Comment (tudor-malene) [inline go/obscuronode/enclave/state.go:118]: basically this if assumes that the genesis rollup hash will be passed in as a config.
Makes sense

File: go/obscuronode/enclave/storage.go [hunk 2]
```diff
@@ -70,12 +70,25 @@ type Storage interface {
 	FetchSecret() SharedEnclaveSecret
 	// StoreSecret stores a secret in the enclave
 	StoreSecret(secret SharedEnclaveSecret)
+
+	// StoreRollupGenesis stores the rollup genesis
+	StoreRollupGenesis(rol *Rollup)
```
Comment (tudor-malene) [inline go/obscuronode/enclave/storage.go:75]: why not: StoreGenesisRollup/FetchGenesisRollup ?

**Review by tudor-malene — CHANGES_REQUESTED** [review]
Thanks @otherview 
looks good.
One very minor change.

File: go/obscuronode/enclave/storage.go (hunk 2 above)
Comment (otherview) [reply go/obscuronode/enclave/storage.go:75]: Sure 👍

**Pull request merged**